src/waypoint.jsx

```diff
@@ -333,13 +352,30 @@ export default class Waypoint extends React.Component {
    * @return {Object}
    */
   render() {
+    if (this.props.noWrapper) {
+      const child = React.Children.only(this.props.children);
+      const ref = (node) => {
+        this.refElement(node);
+        if (this.props.children.ref) {
+          this.props.children.ref(node);
+        }
+      };
+      return React.cloneElement(child, { ref });
+    }
+
+    if (this.props.children) {
+      return <div ref={this.refElement}>{this.props.children}</div>;
+    }
+
     // We need an element that we can locate in the DOM to determine where it is
     // rendered relative to the top of its context.
     return <span ref={this.refElement} style={{ fontSize: 0 }} />;
   }
 }
 
 Waypoint.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.node, PropTypes.array]),
+  noWrapper: PropTypes.bool,
```

Comment:
My opinion is that having multiple ways to do this will be more confusing, because it it less clear which one to choose and why you should choose one over the other. This also makes the documentation more complicated than it needs to be.

Since we have a good solution in the noWrapper branch, my preference would be to go with that only. I'd like to hear from others like @trotzig and @jmeas.